packages/react-router-dev/vite/plugin.ts

```diff
@@ -1836,32 +1836,39 @@ async function handlePrerender(
     );
   }
 
-  async function prerenderData(
-    handler: RequestHandler,
-    prerenderPath: string,
-    clientBuildDirectory: string,
-    reactRouterConfig: Awaited<ReturnType<typeof resolveReactRouterConfig>>,
-    viteConfig: Vite.ResolvedConfig,
-    requestInit: RequestInit
-  ) {
-    let normalizedPath = `${reactRouterConfig.basename}${
-      prerenderPath === "/"
-        ? "/_root.data"
-        : `${prerenderPath.replace(/\/$/, "")}.data`
-    }`.replace(/\/\/+/g, "/");
-    let request = new Request(`http://localhost${normalizedPath}`, requestInit);
-    let response = await handler(request);
-    let data = await response.text();
-
-    validatePrerenderedResponse(response, data, "Prerender", normalizedPath);
-
-    // Write out the .data file
-    let outdir = path.relative(process.cwd(), clientBuildDirectory);
-    let outfile = path.join(outdir, normalizedPath.split("/").join(path.sep));
-    await fse.ensureDir(path.dirname(outfile));
-    await fse.outputFile(outfile, data);
-    viteConfig.logger.info(`Prerender: Generated ${colors.bold(outfile)}`);
-  }
+  await prerenderManifest(
+    build,
+    clientBuildDirectory,
+    reactRouterConfig,
+    viteConfig
+  );
```

Comment:
With fog of war we now should prerender the `__manifest` as well.  This means that statically prerendered sites will load the _full_ manifest on initial load since there's no way to dynamically fetch based on the query params for only outgoing routes - and the enumerations aren't possible to prerender single-request manifest patches.

The only thing I can think of to enable true "fog of war" with prerendering would be to generate a `.manifest` file per path (like we do for `.data`) and then instead of one request to `__manifest` we'd load one request per outgoing path - but that would be be sub-optimal because pages with many links would end many request in parallel.

I think maybe the way to go is when pre-rendering is enabled - disable the "outgoing links" discovery and just let the first link click load the full `__manifest` and then all subsequent links clicks would match syncronously.